packages/hash/api/src/graph/workspace-types.ts

```diff
@@ -327,6 +372,121 @@ const dummyEntityTypeInitializer = async (graphApi: GraphApi) => {
+  description: "Whether or not something has been archived.",
+  possibleValues: [{ primitiveDataType: "Boolean" }],
+});
+
+const summaryPropertyTypeInitializer = propertyTypeInitializer({
+  namespace: WORKSPACE_ACCOUNT_SHORTNAME,
+  title: "Summary",
+  description: "The summary of the something.",
+  possibleValues: [{ primitiveDataType: "Text" }],
+});
+
+const titlePropertyTypeInitializer = propertyTypeInitializer({
+  namespace: WORKSPACE_ACCOUNT_SHORTNAME,
+  title: "Summary",
+  description: "The title of something.",
+  possibleValues: [{ primitiveDataType: "Text" }],
+});
+
+const indexPropertyTypeInitializer = propertyTypeInitializer({
+  namespace: WORKSPACE_ACCOUNT_SHORTNAME,
+  title: "Index",
+  description:
+    "The (fractional) index indicating the current position of something.",
+  possibleValues: [{ primitiveDataType: "Text" }],
+});
+
+const containLinkTypeInitializer = linkTypeInitializer({
+  namespace: WORKSPACE_ACCOUNT_SHORTNAME,
+  title: "Contain",
+  description: "Containing something.",
+});
+
+const parentLinkTypeInitializer = linkTypeInitializer({
+  namespace: WORKSPACE_ACCOUNT_SHORTNAME,
+  title: "Parent",
+  description: "The parent of something.",
+});
+
+const pageEntityTypeInitializer = async (graphApi: GraphApi) => {
+  /* eslint-disable @typescript-eslint/no-use-before-define */
+
+  const summaryPropertyTypeModel =
+    await WORKSPACE_TYPES_INITIALIZERS.propertyType.summary(graphApi);
+
+  const archivedPropertyTypeModel =
+    await WORKSPACE_TYPES_INITIALIZERS.propertyType.archived(graphApi);
+
+  const titlePropertyTypeModel =
+    await WORKSPACE_TYPES_INITIALIZERS.propertyType.title(graphApi);
+
+  const indexPropertyTypeModel =
+    await WORKSPACE_TYPES_INITIALIZERS.propertyType.index(graphApi);
+
+  const containLinkTypeModel =
+    await WORKSPACE_TYPES_INITIALIZERS.linkType.contain(graphApi);
+
+  const parentLinkTypeTypeModel =
+    await WORKSPACE_TYPES_INITIALIZERS.linkType.parent(graphApi);
+
+  const blockEntityTypeModel =
+    await WORKSPACE_TYPES_INITIALIZERS.entityType.block(graphApi);
+
+  const namespace = WORKSPACE_ACCOUNT_SHORTNAME;
+
+  const title = "Page";
+
+  const pageVersionedUri = generateSchemaUri({
+    namespace,
+    title,
+    kind: "entity-type",
+  });
+
+  /* eslint-enable @typescript-eslint/no-use-before-define */
+
+  return entityTypeInitializer({
+    namespace,
+    title,
+    properties: [
+      {
+        propertyTypeBaseUri: summaryPropertyTypeModel.baseUri,
+        propertyTypeVersionedUri: summaryPropertyTypeModel.schema.$id,
+      },
+      {
+        propertyTypeBaseUri: archivedPropertyTypeModel.baseUri,
+        propertyTypeVersionedUri: archivedPropertyTypeModel.schema.$id,
+      },
+      {
+        propertyTypeBaseUri: titlePropertyTypeModel.baseUri,
+        propertyTypeVersionedUri: titlePropertyTypeModel.schema.$id,
+        required: true,
+      },
+      {
+        propertyTypeBaseUri: indexPropertyTypeModel.baseUri,
```

Comment:
This would go without the property-based index as well

Reply:
https://github.com/hashintel/hash/pull/1081#discussion_r979855048